src/main/play-doc/general/MigrationGuide.md

```diff
@@ -33,18 +33,333 @@ In additional to disabling Elasticsearch via configuration, a service lookup mus
   -Dcontrail.syslog.server.service-locator.enabled=off
 ```
 
-### Service ports
+## Service ports
 
-**The services is now deprecated in lieu of [request acl](RequestAclConfiguration).**
+The services define the protocol, port, and/or path under which your service will be addressed to the outside world on.
 
-The services define the protocol, port, and/or path under which your service will be addressed to the outside world on. For example, if http and port 80 are to be used to provide your services and then the following expression can be used to resolve `/myservice` on:
+**From 1.2 onwards, the service ports are deprecated in lieu of [[request acl|RequestAclConfiguration]].**
 
-```json
-    endpoints        = {
-      "angular-seed-play" = {
-        protocol  = "http"
-        bind-port = 0
-        services  = ["http:/myservice"]
+The [[request acl|RequestAclConfiguration]] in conjunction with [[dynamic proxy configuration|DynamicProxyConfiguration]] provides additional proxy mapping flexibility on top of what was previously provided by the service ports. This will allow operations to have a complete control on the proxy configuration.
+
+### Prerequisite
+
+Specify the ConductR target version in the `build.sbt`
+
+```
+BundleKeys.conductrTargetVersion := ConductrVersion.V1_2
+```
+
+The ConductR target version does not need to be declared in `bundle.conf` specified within bundle configuration override.
+
+### Migrating HTTP-based services
+
+Some HTTP based service definition can be migrated by supplying its [[request acl|RequestAclConfiguration]] equivalent, while others require a customised HAProxy configuration.
+
+#### Migrating services definition
+
+The following HTTP based service definition can be migrated by supplying its [[request acl|RequestAclConfiguration]] equivalent.
+
+##### Services mapped to root path
+
+Given the following example where the service URI was "http:/service-name", ConductR will interpret the first path component (`service-name`) as the service name for the purposes of service lookup. By default ConductR will then remove the first path component when rewriting the request. This means that your application or service will receive everything under the root of `/`.
+
+###### Migrating build.sbt
+
+Modify the `Endpoint` declared within the `BundleKeys.endpoints`.
+
+From:
+
+```
+BundleKeys.endpoints := Map(
+  "endpoint-label" -> Endpoint("http", 0, Set(URI("http:/service-name")))
```

Comment:
I'd use `my-service` instead of `service-name` to indicate that this can be an arbitrary service name. With this change the above description is also better readable:

"Given the following example where the service URI was "http:/my-service", ConductR will interpret the first path component (`my-service`) as the service name.."